### DISCUSSION TITLE
Core Knowledge Base & Public API

### DISCUSSION THREAD

**Discussion by ajar** [TripSit/tripsit-discord-bot#198] [Ideas]
I would like to start breaking some functionality out of TripBot and into its own repository that is core to all things TripSit and accessible to the public. Drug knowledge like dosage metrics, combos, aliases, effects, etc. all would be useful to have available on their own through an API. It also would be useful to have action endpoints for logging doses, setting reminders, handling moderation verbs, etc. We can use both REST and GraphQL. I think we need to figure out what endpoints are needed, flesh out their functionality in a manner that is not platform-specific, and get going on it. Thoughts?

**Comment by LunaUrsa** [comment]
I think this would be great: The current API is hosted off of tripbot. We could really use a separate API for this. We'll need to plan this out cuz a lot of people depend on our API
